openassessment/xblock/staff_area_mixin.py

```diff
@@ -278,18 +282,22 @@ def get_student_info_path_and_context(self, student_username):
         if "example-based-assessment" in assessment_steps:
             example_based_assessment = ai_api.get_latest_assessment(submission_uuid)
 
-        workflow_cancellation = workflow_api.get_assessment_workflow_cancellation(submission_uuid)
-        if workflow_cancellation:
-            workflow_cancellation['cancelled_by'] = self.get_username(workflow_cancellation['cancelled_by_id'])
+        workflow = self.get_workflow_info(submission_uuid=submission_uuid)
```

Comment:
I moved all that logic into the new helper function ``get_workflow_cancellation_info`` that I use below. It allowed me to share the bug fix between the student and staff views.